### ISSUE TITLE
VR Support - a couple features that might help

### ISSUE THREAD

**Issue by gorgatron1** [syd711/vpin-studio#555]
**Is your feature request related to a problem? Please describe.**

I recently bought a VR headset and there are a few features that would help me manage it.  The application (a Popper add-on) is described here and the second video is about setting it up:

- https://www.nailbuster.com/wikipinup/doku.php?id=poppervr_config
- https://www.youtube.com/watch?v=qJ7-0rQCmSI

One of the recommendations is that you make a new emulator type, e.g. Visual Pinball VR, because it needs custom start/stop scripts to run.  You have to set this up in Popper Config (fine) but after that there are some things that it would be great if they could be done in VPin Studio.  I still have a cabinet and so I want to be able to manage a mixed set of VR and cabinet tables -- many of them will be the same tables, just a second copy.

Specifically I would like to pick a table and make a duplicate with a new name and a new emulator id.  How that is done I don't care too much.  It could be a duplicate & set emulator in one step or a duplicate and then an edit step.

This is how it looks in Popper Config:

- be able to set the emulator type -- I am not sure how this is determined right now, but I would like to be able to pick it from a list (actually I only care about setting it to this new one)

<img width="1072" alt="image" src="https://github.com/user-attachments/assets/add04e84-2248-468d-b7d7-ebe217cc1e5c">

You can kind of do it here, but I don't want to re-import the table (ideally):

<img width="510" alt="image" src="https://github.com/user-attachments/assets/a850fcc0-bd96-42c1-aaad-73ef25486390">

As I mentioned I want to duplicate a table and give it an alternate name, e.g. duplicate The Addams Family (Bally 1992) G5k 2.4.41 -> The Addams Family [VR] (Bally 1992) G5k 2.4.41 (and copy its associated files)

A workaround (I think) is to mount the drive of my cabinet and upload the tables that are already present.

**Comment by gorgatron1** [comment]
Actually I am not sure what re-uploading an existing file as a clone did ... I didn't see a new table created.

**Comment by gorgatron1** [comment]
I take that back -- I put it in a VR folder and it actually did re-import with the new emulator, but it didn't actually clone the previous.

**Comment by gorgatron1** [comment]
It looks like there is trouble in posting back to VPin Studio what popper is doing -- I am not sure about line 37 but the rest of the curl call looks like the one in the startup:

<img width="448" alt="image" src="https://github.com/user-attachments/assets/ef97767d-0b5a-4de3-b02b-2262ae4ca2ec">

OK, that is the VPin Studio addition to the launch script for the new emulator type, see option 1 on this page:

- https://www.nailbuster.com/wikipinup/doku.php?id=vpx_vr_setup

It is an `;ahk` file -- I think it may not be a normal batch file.  It looks like `Run,curl ...` would do it in this file.

**Comment by gorgatron1** [comment]
After messing with this some more I might just use 2d on my cabinet in popper.  I have a batch script I can run to switch the default VPX binary from normal (DX11) to GL (for VR).

Duplicating a table might still be interesting, but importing the table from the mounted tables directory did work (as long as I didn't use a subfolder).

**Comment by syd711** [comment]
> So that might be a bug: if you import & clone it doesn't duplicate the assets properly if you put it in a subfolder.

Yes, I reproduced and fixed that, will be shipped with the next release.

**Comment by syd711** [comment]
So the actual task that can be resolved here is a "clone to emulator" action for tables, so that they can be setup for VR, right?

**Comment by gorgatron1** [comment]
> So the actual task that can be resolved here is a "clone to emulator" action for tables, so that they can be setup for VR, right?

Yes, though based on my experience I am not bothering with that -- maintaining two sets of tables is too much trouble (but I am a table hoarder so my experience may be atypical).  Anyway, that (have a version of the tables for VR) is the recommended approach per these various videos and write-ups.

I ended up using some bat files to swap out the PinballX.ini and the .starter files -- put the machine in cabinet or VR mode.
